commons/src/main/java/org/infinispan/commons/api/functional/Closeables.java

```diff
@@ -0,0 +1,26 @@
+package org.infinispan.commons.api.functional;
+
+import java.util.Spliterator;
+
+// TODO: Remove once Will's branch closeable spliterator is in.
+
+/**
+ * @since 8.0
+ */
+public class Closeables {
+
+   private Closeables() {
+      // Cannot be instantiated, it's just a holder class
+   }
+
+   /**
+    * Auto closeable spliterator. Being able to close an spliterator is useful
+    * for situations where the iterator might consume considerable resources,
+    * e.g. in a distributed environment. Hence, being able to close it allows
+    * for resources to be reclaimed if the iterator won't be iterated any more.
+    */
+   public interface CloseableSpliterator<T> extends Spliterator<T>, AutoCloseable {
```

Comment:
@danberindei I am guessing you meant to ask me?  Spliterator from my knowledge is not interacted with very often, it is normally used in conjunction with a stream, more specifically a parallel stream.

I found the oracle guides to be decent to start off with [1] [2]

[1] http://www.oracle.com/technetwork/articles/java/ma14-java-se-8-streams-2177646.html
[2] http://www.oracle.com/technetwork/articles/java/architect-streams-pt2-2227132.html